plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -2183,4 +2206,17 @@ public void testAggregationPushdownWithArrays()
                         "(56, VARCHAR 'string_8400', BIGINT '1')," +
                         "(1000, VARCHAR 'string1_8401', BIGINT '1')");
     }
+
+    @Test
+    public void testJson()
+    {
+        assertThat(query("SELECT string_col, json_col, updated_at_seconds FROM " + JSON_TYPE_TABLE))
+                .isFullyPushedDown();
+        assertThat(query("SELECT string_col, json_col, updated_at_seconds" +
+                "  FROM \"SELECT string_col, json_col, updated_at_seconds FROM " + JSON_TYPE_TABLE + "\""))
+                .isFullyPushedDown();
+        assertThat(query("SELECT name FROM \"SELECT json_extract_scalar(json_col, '$.name', 'STRING', '0') AS name" +
+                "  FROM json_table WHERE json_extract_scalar(json_col, '$.id', 'INT', '0') = '1'\""))
```

Comment:
I would recommend testing with normal query instead of dynamic table to improve the readability.

Reply:
I added a similar normal query but left this one in to verify that passthrough queries work as well. lmk if you think I should remove it.